game-core/src/main/java/games/strategy/engine/data/GameMap.java

```diff
@@ -104,6 +104,11 @@ public Territory getTerritory(final String s) {
     return neighbors;
   }
 
+  public Set<Territory> getNeighbors(
+      final Territory territory, @Nullable final Predicate<Territory> cond) {
+    return getNeighbors(territory, cond, Matches.alwaysBi());
```

Comment:
Please rename `Matches.alwaysBi()`. I had a lot of trouble reading that at first, it was not clear what it meant (even the type being passed was not clear, so it looked like 'Bi' was an abbreviation for something').

The implementation for an always true `bi-predicate` is kinda trivial, I'm not sure we really need a matches for it and can just inline it. I don't think we should be encouraging usages of tautology functions by having it be in `Matches`. A tautology funtion makes methods look a lot more complex then they actually are.

Reply:
@DanVanAtta, do you also want to get rid of `Matches.always()`.  I thought that it would be better to have something similar for a `bi-predicate`.

Reply:
Thanks for asking - `Matches.always()` might not be best to inline just yet. Eventually perhaps. Hiding the always true predicate in the implementation class I think is the thing to do. If we can, also hide additional predicates in that class so the callers are just invoking a function and not passing in a predicate. The predicate object is a bit too low level. Consider for example using a method that iterates over territories and requires a predicate, you really have to look at the implementation details to understand what the predicate is going to do exactly. Then if we change that implementation all of the expectations of the callers are broken. I think it's an example of relatively tight coupling and lack of information hiding.

TL;DR - simple inlining of `Matches.always()` could do more harm than good if it is used pervasively. Instead we should seek to first hide as much of the predicate logic in the implementation code to the point where the predicate is an implementation detail, at that time inlining it is likely appropriate.